src/CommunityToolkit/Xamarin.CommunityToolkit/Views/MediaElement/Tizen/MediaPlayerImpl.tizen.cs

```diff
@@ -0,0 +1,438 @@
+			}
+		}
+
+		public void Stop()
+		{
+			cancelToStart = true;
+			_ = ChangeToIdleState();
+			PlaybackStopped?.Invoke(this, EventArgs.Empty);
+		}
+
+		public void SetDisplay(IVideoOutput? output)
+		{
+			VideoOutput = output;
+		}
+
+		public async Task<int> Seek(int ms)
+		{
+			try
+			{
+				await player.SetPlayPositionAsync(ms, true);
+			}
+			catch (Exception e)
+			{
+				Log.Error($"Fail to seek : {e.Message}");
+			}
+			return Position;
+		}
+
+		public void SetSource(Core.MediaSource source)
+		{
+			this.source = source;
+			if (HasSource && AutoPlay)
+			{
+				_ = Start();
+			}
+			else if (!HasSource)
+			{
+				Stop();
+			}
+		}
+
+		public async Task<Stream?> GetAlbumArts()
+		{
+			if (player.State == PlayerState.Idle)
+			{
+				if (tcsForStreamInfo == null || tcsForStreamInfo.Task.IsCompleted)
+				{
+					tcsForStreamInfo = new TaskCompletionSource<bool>();
+				}
+				await tcsForStreamInfo.Task;
+			}
+			await TaskPrepare;
+
+			var imageData = player.StreamInfo.GetAlbumArt();
+			if (imageData == null)
+				return null;
+			return new MemoryStream(imageData);
+		}
+
+		public async Task<IDictionary<string, string>> GetMetadata()
+		{
+			if (player.State == PlayerState.Idle)
+			{
+				if (tcsForStreamInfo == null || tcsForStreamInfo.Task.IsCompleted)
+				{
+					tcsForStreamInfo = new TaskCompletionSource<bool>();
+				}
+				await tcsForStreamInfo.Task;
+			}
+			await TaskPrepare;
+
+			var metadata = new Dictionary<string, string>
+			{
+				[nameof(StreamMetadataKey.Album)] = player.StreamInfo.GetMetadata(StreamMetadataKey.Album),
+				[nameof(StreamMetadataKey.Artist)] = player.StreamInfo.GetMetadata(StreamMetadataKey.Artist),
+				[nameof(StreamMetadataKey.Author)] = player.StreamInfo.GetMetadata(StreamMetadataKey.Author),
+				[nameof(StreamMetadataKey.Genre)] = player.StreamInfo.GetMetadata(StreamMetadataKey.Genre),
+				[nameof(StreamMetadataKey.Title)] = player.StreamInfo.GetMetadata(StreamMetadataKey.Title),
+				[nameof(StreamMetadataKey.Year)] = player.StreamInfo.GetMetadata(StreamMetadataKey.Year)
+			};
+			return metadata;
+		}
+
+		public async Task<MSize> GetVideoSize()
+		{
+			if (player.State == PlayerState.Idle)
+			{
+				if (tcsForStreamInfo == null || tcsForStreamInfo.Task.IsCompleted)
+				{
+					tcsForStreamInfo = new TaskCompletionSource<bool>();
+				}
+				await tcsForStreamInfo.Task;
+			}
+			await TaskPrepare;
+
+			var videoSize = player.StreamInfo.GetVideoProperties().Size;
+			return new MSize(videoSize.Width, videoSize.Height);
+		}
+
+		public View GetEmbeddingControlView(IMediaPlayer player)
+		{
+			return new EmbeddingControls
+			{
+				BindingContext = player
+			};
+		}
+
+		public void Dispose()
+		{
+			Dispose(true);
+			GC.SuppressFinalize(this);
+		}
+
+		protected virtual void Dispose(bool disposing)
+		{
+			if (disposed)
+				return;
+
+			if (disposing)
+			{
+				player.PlaybackCompleted -= OnPlaybackCompleted;
```

Comment:
Are the `Tizen` objects are unmanaged resources in dotnet world?

Reply:
yes..